### PR TITLE
Create a plugin for additional filetypes (such as parquet)

### PR DESCRIPTION
**Pull request by Aggrathon**
This creates a new plugin for xiplot that (currently) adds parquet and feather (not in WASM) support.
The support for those formats is removed from xiplot, but the plugin is added as a (non-WASM) dependency.
The plugin is in the same repo since it is literally just two files (one `__init__.py` and one `pyproject.toml`).

(I'll update the WASM branch to have this as a recommended plugin, if the plugin seems like a good idea)

### CODE REVIEW

File: xiplot/utils/dataframe.py [hunk 1]
```diff
@@ -93,16 +62,14 @@ def write_functions() -> (
         mime: MIME type of the written data.
     """
     for _, _, plugin in get_plugins_cached("write"):
-        yield plugin()
+        plugin = plugin()
+        if plugin:
```
Comment (juntyr) [inline xiplot/utils/dataframe.py:66]:
```suggestion
        if plugin is not None:
```

File: xiplot/utils/dataframe.py [hunk 2]
```diff
@@ -61,7 +32,9 @@ def read_functions() -> (
         ext: File extension that the readed can handle.
     """
     for _, _, plugin in get_plugins_cached("read"):
-        yield plugin()
+        plugin = plugin()
+        if plugin:
```
Comment (juntyr) [inline xiplot/utils/dataframe.py:36]:
```suggestion
        if plugin is not None:
```

File: docs/user_guide/data_files.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ The plots and the data file will be downloaded as a .tar file. It can be restore
 
 ## Dataframe loading and saving
 
-`xiplot` uses `pandas` to load dataframes from `.csv`, `.json`, `.ft` (feather format) and `.parquet` files. If you are running `xiplot` locally, you can simply copy your datasets into the `data/` folder. If you are accessing `xiplot` remotely or you do not want to pollute the `data/` folder, you can upload the dataset into memory directly inside `xiplot`: navigate to the data tab and either click the upload button to the right or drag the file into it.
+`xiplot` uses `pandas` to load dataframes from `.csv`, `.json`, `.feather`, and `.parquet` files. If you are running `xiplot` locally, you can simply copy your datasets into the `data/` folder. If you are accessing `xiplot` remotely or you do not want to pollute the `data/` folder, you can upload the dataset into memory directly inside `xiplot`: navigate to the data tab and either click the upload button to the right or drag the file into it.
```
Comment (juntyr) [inline docs/user_guide/data_files.md:49]: Could we feather and parquet mentions here have a footnote that tells the user that they require the new plugin, which is pre-installed in the local version but not in WASM?

File: plugin_xiplot_filetypes/xiplot_filetypes/__init__.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+from io import BytesIO
+
+import pandas as pd
+
+
+def read_parquet():
+    try:
```
Comment (juntyr) [inline plugin_xiplot_filetypes/xiplot_filetypes/__init__.py:7]: Why do we expect the import to fail, even when the plugin is installed?

File: pyproject.toml [hunk 5]
```diff
@@ -72,3 +69,5 @@ profile = "black"
 line_length = 79
 
 [tool.flake8]
+max-line-length = 88
```
Comment (juntyr) [inline pyproject.toml:72]: Why do black and isort use different line length values above?

File: plugin_xiplot_filetypes/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# [&chi;iplot](https://github.com/edahelsinki/xiplot) plugin for additional file types
+
+This plugin adds support for additional file types (beside `csv` and `json`) to [&chi;iplot](https://github.com/edahelsinki/xiplot).
+Currently, this plugin adds support for `feather` and `parquet`.
+Note that in the [WASM version](https://edahelsinki.fi/xiplot) only `parquet` is added.
```
Comment (juntyr) [inline plugin_xiplot_filetypes/README.md:5]:
```suggestion
Note that in the [WASM version](https://edahelsinki.fi/xiplot) only support for `parquet` is added.
```

Comment (juntyr) [inline plugin_xiplot_filetypes/README.md:5]: There has been some recent activity in https://github.com/pyodide/pyodide/issues/2933, so getting pyarrow into Pyodide doesn’t seem too far away anymore. So while this restriction is ok for now, we should keep an eye on that issue and remove the limitation once possible.

**Comment by juntyr** [comment]
With the suggested minor tweaks, this looks good to me.

File: plugin_xiplot_filetypes/README.md (hunk 6 above)
Comment (Aggrathon) [reply plugin_xiplot_filetypes/README.md:5]: nice

File: plugin_xiplot_filetypes/xiplot_filetypes/__init__.py (hunk 4 above)
Comment (Aggrathon) [reply plugin_xiplot_filetypes/xiplot_filetypes/__init__.py:7]: Yeah, the parquet one should never fail. Mostly a leftover from copy pasting (and maybe as a pattern for future file types).

File: plugin_xiplot_filetypes/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# [&chi;iplot](https://github.com/edahelsinki/xiplot) plugin for additional file types
+
+This plugin adds support for additional file types (beside `csv` and `json`) to [&chi;iplot](https://github.com/edahelsinki/xiplot).
+Currently, this plugin adds support for `feather` and `parquet`.
+Note that in the [WASM version](https://edahelsinki.fi/xiplot) only support for `parquet` is added.
```
Comment (juntyr) [inline plugin_xiplot_filetypes/README.md:5]: Perhaps this could also have a brief explainer for how to install the plugin in WASM?

**Pull request merged**